salt/cloud/__init__.py

```diff
@@ -234,7 +234,7 @@ def _opts_defaults(self, **kwargs):
                          if a.get('provider', '')]
             if providers:
                 _providers = opts.get('providers', {})
-                for provider in list(_providers):
+                for provider in list(_providers).copy():
```

Comment:
I'm not sure about the entirety of this data structure but are you sure you don't want a `deepcopy()` here? (I'm not saying this is the case, but just to raise the point to make sure it's been considered.) :]

Reply:
we are doing a copy on the list, which is just the keys, which can't be complex data structures, so we should be fine.

Reply:
It's not a very complex data structure. I don't think we'll need a `deepcopy()` here.